data_dictionary/keyspace_element.hh

```diff
@@ -41,4 +46,29 @@ public:
     virtual std::ostream& describe(replica::database& db, std::ostream& os, bool with_internals) const { return describe(os); }
 };
 
-}
+/**
+ * Common interface for types able to generate multiple descriptions
+ * of the same type (sharing the same `keyspace_name` and `element_type`).
+ *
+ * Analogous to `data_dictionary::keyspace_element`.
+ */
+class keyspace_element_generator {
```

Comment:
This tells the reader this class generates keyspace_element objects. But it doesn't, it describes them.